### PR TITLE
RSLC- 9 -Target scheduler

### PR DESCRIPTION
**Pull request by RCarmona53**
#### Board:
* [RSLC-9](https://trello.com/c/RVz0DUwz/10-rslc-9-as-a-system-i-should-delete-targets-after-one-week)
---
#### Description:
As a system, I should delete targets after one week

### CODE REVIEW

**Review by santiagodiaz — APPROVED** [review]
great! 🚀

File: app/jobs/cleanup_job.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class CleanupJob < ApplicationJob
+  queue_as :default
+
+  def perform
+    Target.where('created_at <= ?', 1.week.ago).destroy_all
```
Comment (enzofab91) [inline app/jobs/cleanup_job.rb]: As a suggestion, this kind of things can be done in a scope. See https://guides.rubyonrails.org/active_record_querying.html#scopes

File: spec/jobs/cleanup_job_spec.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe CleanupJob, type: :job do
+  let!(:user) { create(:user) }
+  let!(:targets) { create_list(:target, 2, user:) }
+  let!(:old_target) { create(:target, user:, created_at: 2.weeks.ago) }
+
+  it 'deletes targets older than one week' do
```
Comment (enzofab91) [inline spec/jobs/cleanup_job_spec.rb:8]: You can also validate that the target that is actually being deleted is the `old_target` and not the others. Something like

expect(Target.all).to include(active_target)

```

And something similar to check that the active targets are still alive.
it 'only delete old targets' do
   subject
   expect(Target.all).not_to include(old_target)
end
```

**Review by enzofab91 — APPROVED** [review]
Nice work! 👏 
Left some non blockers comments.

File: app/jobs/cleanup_job.rb (hunk 1 above)
Comment (RCarmona53) [reply app/jobs/cleanup_job.rb]: Done! Let me know if like this its ok, or there is something to improve. Thanks!

File: spec/jobs/cleanup_job_spec.rb (hunk 2 above)
Comment (RCarmona53) [reply spec/jobs/cleanup_job_spec.rb:8]: Done!

File: app/jobs/cleanup_job.rb (hunk 1 above)
Comment (enzofab91) [reply app/jobs/cleanup_job.rb]: I think you forgot to commit the changes 🤔

File: app/jobs/cleanup_job.rb [hunk 3]
```diff
@@ -2,6 +2,6 @@ class CleanupJob < ApplicationJob
   queue_as :default
 
   def perform
-    Target.where('created_at <= ?', 1.week.ago).destroy_all
+    Target.cleanup_old_targets
```
Comment (enzofab91) [inline app/jobs/cleanup_job.rb:5]:
```suggestion
    Target.older_than_one_week.destroy_all
```

**Review by enzofab91 — APPROVED** [review]
LGTM

File: app/jobs/cleanup_job.rb (hunk 1 above)
Comment (RCarmona53) [reply app/jobs/cleanup_job.rb]: Mb now should appear the changes.

**Pull request merged**